src/Tomlyn.Tests/FloatingRoundtripTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Alexandre Mutel. All rights reserved.
+// Licensed under the BSD-Clause 2 license.
+// See license.txt file in the project root for full license information.
+
+using NUnit.Framework;
+using Tomlyn.Model;
+
+namespace Tomlyn.Tests
+{
+    [TestFixture]
+    public class FloatingRoundtripTests{
+
+
+        [TestCase(0.0)]
+        [TestCase(-0.0)]
+        [TestCase(1.0)]
+        [TestCase(-1.0)]
+        [TestCase(double.PositiveInfinity)]
+        [TestCase(double.NegativeInfinity)]
+        [TestCase(double.NaN)]
+#if NET8_0_OR_GREATER
+        [TestCase(double.NegativeZero)]
+#endif
+        [TestCase(double.Epsilon)]
+        [TestCase(-double.Epsilon)]
+        // [TestCase(0.1)] - - These fail. Adjusting g16->g17 fixes some but not all.
```

Comment:
Indeed, I remember that I had a problem with double and I had to use g16 instead of g17, but I didn't dig further.

Reply:
Yeah, I'm not sure what the best approach is there. Or for f32. But it's easier to explore, now. One thing I didn't add was testing the string representation, such as if we got weird 0000001 suffixes etc.